src/Analyzers/CSharp/Analyzers/UseImplicitObjectCreation/CSharpUseImplicitObjectCreationDiagnosticAnalyzer.cs

```diff
@@ -93,18 +97,46 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
                     _ => null,
                 };
             }
+            else if (objectCreation.Parent.IsKind(SyntaxKind.ArrayInitializerExpression) &&
+                objectCreation.Parent.Parent is ArrayCreationExpressionSyntax arrayCreation)
+            {
+                typeNode = arrayCreation.Type.ElementType;
+            }
+            else if (objectCreation.Parent.IsKind(SyntaxKind.CollectionInitializerExpression) &&
+                objectCreation.Parent.Parent is ObjectCreationExpressionSyntax collectionObjectCreation)
+            {
+                var collectionTypeSymbol = semanticModel.GetTypeInfo(collectionObjectCreation, cancellationToken).Type;
+                var targetTypeSymbol = semanticModel.GetTypeInfo(objectCreation, cancellationToken).ConvertedType;
+
+                if (collectionTypeSymbol != null && targetTypeSymbol != null)
```

Comment:
i don't think we want to do this here.  The point of "is apparent" is that something in the code itself was explicit about what the type was.  So, similar to the array creation case, i think we want to look directly at the type syntax node of hte object-creation expression, and get it's type-argument.

We then also want to ensure that that type-argument matches the collection type.  If so, we can use that typenode.

Reply:
Understood. I was going too general with this before we talked about the "is apparent" idea. Will revert back.